src/lib/shim/src/lib.rs

```diff
@@ -44,18 +45,56 @@ pub fn simtime() -> Option<SimulationTime> {
     SimulationTime::from_c_simtime(unsafe { bindings::shim_sys_get_simtime_nanos() })
 }
 
-mod tls_allow_native_syscalls {
-    use super::*;
+/// Describes whether something is within the context of the shadow (shim) code,
+/// or the application/plugin code.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, IntoPrimitive, TryFromPrimitive)]
+#[repr(u8)]
+pub enum ExecutionContext {
+    Shadow,
+    Application,
+}
+
+static CURRENT_EXECUTION_CONTEXT: ShimTlsVar<Cell<ExecutionContext>> =
+    ShimTlsVar::new(&SHIM_TLS, || Cell::new(ExecutionContext::Application));
```

Comment:
Maybe add a comment to this saying not to access it directly, and only access it through `ExecutionContext`?

Reply:
Done.